app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/navigationui/StoreHistoryTask.java

```diff
@@ -0,0 +1,60 @@
+package com.mapbox.services.android.navigation.testapp.activity.navigationui;
+
+import android.os.AsyncTask;
+import android.os.Environment;
+
+import com.mapbox.services.android.navigation.v5.navigation.MapboxNavigation;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStreamWriter;
+
+import timber.log.Timber;
+
+class StoreHistoryTask extends AsyncTask<Void, Void, Void> {
+  private static final String EMPTY_HISTORY = "{}";
```

Comment:
when you call `getHistory` it will return a mostly empty history but i think it will at least have a few things. i don't think it will return just `{}`. it should be something like:

```json
{
  "version": "foo.bar",
  "events":[]
}
```

Reply:
I've seen `{}` while testing, that's why I added that check - so we don't mess it up with replacing the whole history with an empty JSON.